File: LanguageTutorials/DesignPatterns/src/main/java/org/bytescale/patterns/observer/WeatherData.java
package org.bytescale.patterns.observer;

import java.util.ArrayList;
import java.util.Observable;

public class WeatherData extends Observable {
    private ArrayList<java.util.Observer> observers;
    private float temperature;
    private float humidity;
    private float pressure;

    public WeatherData() {

    }

    public void measurementsChanged() {
        setChanged();
        notifyObservers();
    }

    public void setMeasurements(float temperature, float humidity, float pressure) {
        this.temperature = temperature;
        this.humidity = humidity;
        this.pressure = pressure;
        measurementsChanged();
    }

    public float getTemperature() {
        return temperature;
    }

    public float getHumidity() {
        return humidity;
    }

    public float getPressure() {
        return pressure;
    }


}

File: LanguageTutorials/CSDesignPatterns/Org.Bytescale.Patterns/Observer/WeatherData.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Org.Bytescale.Patterns.Observer
{
    public class WeatherData
    {
        private float _temperature;
        private float _humidity;
        private float _pressure;

        public void SetMeasurements(float temperature, float humidity, float pressure)
        {
            _temperature = temperature;
            _humidity = humidity;
            _pressure = pressure;

        }

        public float Temperature { get { return _temperature; } }
        public float Humidity { get { return _humidity; } }
        public float Pressure { get { return _pressure; } }


    }
}
